pysoa/server/server.py

```diff
@@ -557,8 +561,12 @@ def execute_job(self, job_request):  # type: (EnrichedJobRequest) -> JobResponse
                 if action_in_class_map:
                     action = self.action_class_map[action_request.action](self.settings)
                 elif action_request.action == 'introspect':
-                    from pysoa.server.action.introspection import IntrospectionAction
-                    action = IntrospectionAction(server=self)
+                    # If set, use custom introspection action. Use default otherwise.
+                    if self.introspection_action is not None:
+                        action = self.introspection_action(server=self)
+                    else:
+                        from pysoa.server.action.introspection import IntrospectionAction
```

Comment:
I've run into problems with local imports like this, because they're not run every time the code is.  e.g. I encountered a broken import that hadn't been found because it was in a branch for an error case.  Is there any harm in importing this at the top level?

Reply:
Unfortunately, in that case `pysoa.server.server` and `pysoa.server.action.introspection` would import each other. This causes an import loop and the `pysoa.server.server` module fails to load. This could be solved by moving (a substantial amount) of code around in PySOA, but it would also break some import paths.